### ISSUE TITLE
Room names in Russian: Codepage issues?

### ISSUE THREAD

**Issue by nokia-lmt** [Juerd/eq3-max#12]
Good day, 

All my rooms are named in Russian and all such labels are corrupted. (I suspect EQ3 Cube does not support UTF8?)  See attached screenshot. Is there is a way to support non-latin characters? Thank you!

![eq3 error](https://user-images.githubusercontent.com/59649157/142729121-b156ae86-4177-4c72-90a4-8a4799a25ea8.JPG)

**Comment by Juerd** [comment]
Can you provide the output of a hex dump (which you can make with `max | hd`) please, at least the part where one room name should be? And can you provide the name of that room in proper cyrillic?

It looks like double encoded utf8 so it might very well be a bug or wrong assumption in this project, but I'd like to verify that first.

**Comment by nokia-lmt** [comment]
See attached please. I can't uderstand where to look for room names in such dump, sorry.
![MAX-HD](https://user-images.githubusercontent.com/59649157/148078690-f3f19e1b-11e5-4375-aa17-7b73532cf608.JPG)

**Comment by Juerd** [comment]
Is the first room name Детскач? I can't read your language so I have no idea if that's correct. But if it is, a fix will be easy :)

**Comment by nokia-lmt** [comment]
Hi Juerd,

Yes, you're almost correct. The name of this room is Детская. If you can provide a hotfix, I'm ready to test it ASAP. Thanks in advance!

**Comment by Juerd** [comment]
Sorry, typo. Should be Детская (and it's only the first word).

Google says it's a word, so I'll assume this is right.

**Comment by nokia-lmt** [comment]
Yes, as I said this is a correct name of the room. I'm awaiting promised hotfix. Thanks in advance! Appreciate it.

**Comment by Juerd** [comment]
In `lib/Max.pm` add the following line:

     utf8::decode($name);

After the following two lines:

     $offset += 5 + length $name;
     and
     $offset += 16 + length $name;

This is an incomplete fix (don't set any room names using `max name` because that will likely break) but should suffice for testing if I'm on the right track.

**Comment by nokia-lmt** [comment]
Just applied changes and tested them - works perfectly! When to expect a complete version of the hotfix?

![Max](https://user-images.githubusercontent.com/59649157/148087364-d81c39c8-f2db-4de4-bb53-2bd7443fa024.JPG)

**Comment by Juerd** [comment]
Glad that works! I don't know when I will get to fixing it.

**Comment by nokia-lmt** [comment]
Well noted, Juerd. In either case please keep me posted. Thank you!
